spec/core/v2/ics-004-channel-and-packet-semantics/README.md

```diff
@@ -686,47 +703,65 @@ function recvPacket(
+    acknowledgement,success = cbs.onReceivePacket(packet.destChannel,packet.sourceChannel,packet.sequence,payload,relayer) // Note that payload includes the version. The application is required to inspect the version to route the data to the proper callback
     abortTransactionUnless(success)
     if ack != nil {
         // NOTE: Synchronous ack. 
-        writeAcknowledgement(packet.channelDestId,packet.sequence,ack)
-        // In case of Synchronous ack we emit the event here as we have all the necessary information, while writeAcknowledgement can only retrieve this in case of asynchronous ack. 
-        emitLogEntry("writeAcknowledgement", {
-            sequence: packet.sequence,
-            sourceId: packet.channelSourceId,
-            destId: packet.channelDestId,
-            timeoutTimestamp: packet.timeoutTimestamp,
-            data: packet.data,
-            ack
-        })
+        writeAcknowledgement(packet.destChannel,packet.sequence,ack)
+        // emit an acknowledgement event for each app acknowledgement in the acknowledgement
+        for i, ack in acknowledgment {
+            // In case of Synchronous ack we emit the event here as we have all the necessary information, while writeAcknowledgement can only retrieve this in case of asynchronous ack. 
+            emitEvents("write_acknowledgement", {
+                sourceChannel: packet.sourceChannel,
+                destChannel: packet.destChannel,
+                sequence: packet.sequence, // value is string in decimal format
+                payloadSequence: i, // value is string in decimal format
+                acknowledgement: toHex(ack),
+            })
+        }
     }else {
     // NOTE No ack || Asynchronous ack. 
     // ack is nil and will be written asynchronously, so we store the full packet in the private store
-        storedPacket[packet.channelDestId,packet.sequence]=packet
+        storedPacket[packet.destChannel,packet.sequence]=packet
     }
     // Provable Stores 
     // we must set the receipt so it can be verified on the other side
     // it's the sentinel success receipt: []byte{0x01}
     provableStore.set(
-        packetReceiptPath(packet.channelDestId, packet.sequence),
+        packetReceiptPath(packet.destChannel, packet.sequence),
         SUCCESSFUL_RECEIPT
     )
 
-    // log that a packet has been received
-    // Event Emission
-    emitLogEntry("recvPacket", {
-      data: packet.data
-      timeoutTimestamp: packet.timeoutTimestamp,
-      sequence: packet.sequence,
-      sourceId: packet.channelSourceId,
-      destId: packet.channelDestId,
-      relayer: relayer 
+    // Event Emission for receive packet
+    emitEvents("recv_packet", {
+      sourceChannel: sourceChannelId, 
+      destChannel: channel.counterpartyChannelId,
```

Comment:
meant to drop comment here, counterparty channel id on recv will be equal to source channel no? Should be `packet.DestChannel`?